### PR TITLE
[BTFS-928]for v0.2.3 release: switch to grpc client

### PR DESCRIPTION
**Pull request by laipogo**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactoring

* **What is the current behavior?** (You can also link to an open issue here)
send node metrics with http client.

* **What is the new behavior?** (You can also refer to a JIRA ticket here)
send node metrics with grpc client.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
nope

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)
https://github.com/TRON-US/go-btfs-common/pull/15

* **Description of changes**
http client -> grpc client

---

* **Please check if the PR fulfills these requirements**
- [x] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [x] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [x] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [x] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [x] Code changes have run through `go fmt`
- [x] Code changes have run through `go mod tidy`
- [x] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: analytics/analytics.go [hunk 1]
```diff
@@ -158,51 +147,103 @@ func (dc *dataCollection) update() {
...
-		dc.reportHealthAlert(fmt.Sprintf("failed to marshal databag: %s", err.Error()))
-		return
+		return err
 	}
+	defer cancel()
+	defer conn.Close()
 
-	// btfs node reports to status server by making HTTP request
-	req, err := http.NewRequest("POST", fmt.Sprintf("%s%s", statusServerDomain, routeMetrics), bytes.NewReader(dataBagMarshaled))
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (wxue) [inline analytics/analytics.go]: same comment here: "I'd make 5*time.Second a timeout const for future convenient adjustment and avoid the magic number"

**Comment by taiyangc** [comment]
Can you edit to title to include "for v0.2.3 release"

**Comment by taiyangc** [comment]
@laipogo Have you tested this build by sending some stats to a grpc-based status server (maybe locally)?

**Comment by laipogo** [comment]
> @laipogo Have you tested this build by sending some stats to a grpc-based status server (maybe locally)?
yes, i pointed status-server to local, and tried `make install` and `btfs daemon --dc`, status-server received the message and processed correctly.
also i tested the retry feature.

**Pull request merged**